### PR TITLE
Consolidate fork,exec,exit for Linux

### PR DESCRIPTION
**Pull request by nicholasberlin**
## Change Summary

Changes are for aggregation of fork, exec, and exit process events. Which means adding handling for `event.action` being an array. Added so go code, and removed the previous documentation files after building a new consolidated version.

## Release Target

8.16

### For mapping changes:

- [x] I ran `make` after making the schema changes, and committed all changes
- [ ] If these field(s) are "exception"-able, I made a companion PR to Kibana adding it (see [Readme](https://github.com/elastic/endpoint-package#exceptionable))
- [ ] If this is a `metadata` change, I also updated both transform destination schemas to match

### CODE REVIEW

File: custom_documentation/doc/endpoint/process/linux/linux_process_exit.md
Comment (nicholasberlin) [inline custom_documentation/doc/endpoint/process/linux/linux_process_exit.md:1]: This was deleted in favor the consolidated file `custom_documentation/doc/endpoint/process/linux/linux_process_fork_exec_and_exit.md`

File: custom_documentation/doc/endpoint/process/linux/linux_process_fork.md
Comment (nicholasberlin) [inline custom_documentation/doc/endpoint/process/linux/linux_process_fork.md:1]: This was deleted in favor the consolidated file `custom_documentation/doc/endpoint/process/linux/linux_process_fork_exec_and_exit.md`

File: custom_documentation/doc/endpoint/process/linux/linux_process_fork_exec_exit.md
Comment (nicholasberlin) [inline custom_documentation/doc/endpoint/process/linux/linux_process_fork_exec_exit.md:1]: This file was `git mv` and updated with the latest changes.

File: custom_documentation/src/endpoint/data_stream/process/linux/linux_process_fork_exec_exit.yaml
Comment (nicholasberlin) [inline custom_documentation/src/endpoint/data_stream/process/linux/linux_process_fork_exec_exit.yaml:1]: This file was `git mv` and updated with the latest changes.

File: custom_documentation/src/endpoint/data_stream/process/linux/linux_process_exec.yaml
Comment (nicholasberlin) [inline custom_documentation/src/endpoint/data_stream/process/linux/linux_process_exec.yaml:1]: This was deleted in favor the consolidated file `custom_documentation/src/endpoint/data_stream/process/linux/linux_process_fork_exec_and_exit.yaml`

File: custom_documentation/src/endpoint/data_stream/process/linux/linux_process_fork.yaml
Comment (nicholasberlin) [inline custom_documentation/src/endpoint/data_stream/process/linux/linux_process_fork.yaml:1]: This was deleted in favor the consolidated file `custom_documentation/src/endpoint/data_stream/process/linux/linux_process_fork_exec_and_exit.yaml`

File: scripts/generate-docs/custom_doc.go [hunk 1]
```diff
@@ -30,10 +30,28 @@ type eventOverview struct {
...
 
+type StringArray []string
+
 type eventIdentification struct {
-	Os         []string          `yaml:"os"`
-	DataStream string            `yaml:"data_stream"`
-	Filter     map[string]string `yaml:"filter"`
+	Os         []string               `yaml:"os"`
+	DataStream string                 `yaml:"data_stream"`
+	Filter     map[string]StringArray `yaml:"filter"`
```
Comment (nicholasberlin) [inline scripts/generate-docs/custom_doc.go:38]: This was changed from a string-to-string map to a string-to-[]string map.
And custom unmarshal function was added, and handling for `event.action` unmarshalling to an array was added below.

**Review by pzl — APPROVED** [review]
i'll clear the approval requirement for defend workflows, this does not affect installed package contents or mapping :+1:

**Review by Trinity2019 — APPROVED** [review]

**Pull request merged**